TowerDefense/GameEngine.cs

```diff
@@ -24,41 +20,17 @@ internal GameEngine()
 
         internal void Update(GameTime gameTime)
         {
+            delayedActions.RemoveAll(d => d.Update(gameTime));
```

Comment:
Not sure if this is an anti-pattern, but I have Update set to return true if I want it to be removed from the parent, so my update logic starts to look like this